pkg/controller/rediscluster/checker.go

```diff
@@ -97,20 +96,22 @@ func (r *RedisClusterHandler) CheckAndHeal(meta *clustercache.Meta) error {
 			}
 		}
 	}
-	for _, sip := range sentinels {
-		if err := r.rcChecker.CheckSentinelNumberInMemory(sip, meta.Obj, meta.Auth); err != nil {
-			r.logger.WithValues("namespace", meta.Obj.Namespace, "name", meta.Obj.Name).Info(err.Error())
-			if err := r.rcHealer.RestoreSentinel(sip, meta.Auth); err != nil {
-				return err
-			}
-		}
-	}
+	//for _, sip := range sentinels {
+	//	if err := r.rcChecker.CheckSentinelNumberInMemory(sip, meta.Obj, meta.Auth); err != nil {
+	//		r.logger.WithValues("namespace", meta.Obj.Namespace, "name", meta.Obj.Name).Info(err.Error())
+	//		if err := r.rcHealer.RestoreSentinel(sip, meta.Auth); err != nil {
+	//			return err
+	//		}
+	//		return needRequeueErr
+	//	}
+	//}
 	for _, sip := range sentinels {
 		if err := r.rcChecker.CheckSentinelSlavesNumberInMemory(sip, meta.Obj, meta.Auth); err != nil {
 			r.logger.WithValues("namespace", meta.Obj.Namespace, "name", meta.Obj.Name).Info(err.Error())
 			if err := r.rcHealer.RestoreSentinel(sip, meta.Auth); err != nil {
 				return err
 			}
+			return needRequeueErr
```

Comment:
A hidden logic was imported here: operator check and fix ONE sentinel, and wait for next reconcile loop to fix the the next sentinel. I think operator should do all the work in one reconcile loop.